My project/Assets/Scripts/CreatureLogic/Actions/Breeding/LookForMate.cs

```diff
@@ -0,0 +1,108 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class LookForMate : IAction
+{
+    //Class Variables
+    CreatureData data;
+    Rigidbody2D rb;
+    RangeScanner scanner;
+    int requiredEnergy;
+    //the time when creature can look for mate again
+    //ie: time is 1:20 then creature can only look for mate after 1:20
+    float nextLookForMate = 0;
```

Comment:
same comment as above: remove this variable and derive it from searchTimeEnd + cooldown